### DISCUSSION TITLE
Does lazygit use `--force` or `--force-with-lease` while force pushing commit?

### DISCUSSION THREAD

**Discussion by byteZorvin** [jesseduffield/lazygit#4068] [Q&A] [answered]
When my commit history and lg shows a dialog box if I want to force push the commits, which option is used while force pushing.
It is safer to use `--force-with-lease` which does not override a teammates work and is safer to use.

**Comment by mark2185** [comment]
[Related discussion](https://github.com/jesseduffield/lazygit/issues/1668).

**Comment by stefanhaller** [comment, marked as answer]
> It is safer to use `--force-with-lease`

No, it is not, at least not when used on its own. I have been trying to combat this misinformation for a long time now. When used on its own, `--force-with-lease` provides very little additional protection over `--force`. See the link that Luka posted above for more information.

But yes, lazygit does use `--force-with-lease`, and it is strongly recommended that you _also_ turn on `git config --global push.useForceIfIncludes true` to make it actually safe.

**Reply by byteZorvin** [reply]
I read the above discussion and understood that `--force-with-lease` offer no security. 

> But yes, lazygit does use --force-with-lease

So why not use `--force-if-includes` instead of `--force-with-lease`

**Reply by stefanhaller** [reply]
That's a very good question. Unfortunately, `--force-if-includes` is not perfect and sometimes rejects a push that is perfectly fine; most notably, when the local branch name is different from the remote tracking branch (see [here](https://public-inbox.org/git/f51c73ed-eb03-83ca-fb31-d3e2645c9a63@haller-berlin.de/); unfortunately there were no replies). So some people might not want to turn it on for that reason, so we would have to make it configurable in lazygit. However, it is our policy that if something is already configurable in git, we don't add the same option again in lazygit.

**Reply by byteZorvin** [reply]
Thank for the quick response. Got my answer. Thanks for making lazygit so awesome  ❤️
